package/MDAnalysis/coordinates/DCD.py

```diff
@@ -410,17 +410,36 @@ def __init__(self,
             is_periodic=1,
             istart=istart)
 
-    def write_next_timestep(self, ts):
```

Comment:
I feel that it *might* be useful to leave one entry point open for handing over custom-timesteps. Perhaps not deprecate here?

Or are there better alternative ways to do non-standard things (such as writing a "trajectory" of principal components... someone did this a while ago), e.g., using Universe.empty() and friends?

Reply:
are the low-level functions in `lib.formats` good for that?

Reply:
why is this change in the first place?  This is an internal API method to me. The write method in our base class used to handle this just fine. It also allowed us to have the code for unpacking the atomgroup into a timestep just *once*. I would rather vote for making this a private method and leaving childclasses as they are otherwise.

Reply:
The problem was `Writer.write(obj)` then calls `Writer.write_next_timestep(ts)` (which is what is implemented in child classes).  But really the argument for `W.write_next_timestep()` should still be an `AtomGroup` (else you end up with car crash code [like this](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/coordinates/PDB.py#L884))

So basically more of #206 where it's not really clear what a `Writer` is meant to implement

I guess we could make it so that:
 - Child classes implement `write_next_timestep()` which **must** accept AG/Universe
 - Child classes can optionally accept a `Timestep` in the above method?

Currently the (very loose) default is that `Timestep` is accepted, and `AG` is optional.  (But there are a few dissenters to this)

Reply:
I accept consistency as reasoning. I would still like that this code of unpacking an atomgroup exists only once as a helper function. We can properly test it.